source/common/network/lc_trie.h

```diff
@@ -445,21 +445,13 @@ template <class T> class LcTrie {
       ip_prefixes_ = data;
       std::sort(ip_prefixes_.begin(), ip_prefixes_.end());
 
-      // In theory, the trie_ vector can have at most twice the number of ip_prefixes entries - 1.
-      // However, due to the fill factor a buffer is added to the size of the
-      // trie_. The buffer value(2000000) is reused from the reference implementation in
-      // http://www.csc.kth.se/~snilsson/software/router/C/trie.c.
-      // TODO(ccaraman): Define a better buffer value when resizing the trie_.
-      maximum_trie_node_size = 2 * ip_prefixes_.size() + 2000000;
-      trie_.resize(maximum_trie_node_size);
-
       // Build the trie_.
+      trie_.reserve(size_t(ip_prefixes_.size() / fill_factor_));
       uint32_t next_free_index = 1;
       buildRecursive(0u, 0u, ip_prefixes_.size(), 0u, next_free_index);
 
       // The value of next_free_index is the final size of the trie_.
-      trie_.resize(next_free_index);
-      trie_.shrink_to_fit();
```

Comment:
i don't think that I agree with removing those 2 lines... We still need to shrink it to release the memory, don't we?

Reply:
In the default case where `fill_ratio` is 0.5, it arguably doesn't make sense to release the memory here: the `capacity` of the buffer at this point can potentially be up to 2x the used `size`, but that's true of most vectors if the `std::vector` implementation uses a typical power-of-two allocator internally. For smaller values of `fill_ratio`, though, reclaiming space here might be a win. I'll try reenabling the shrink-to-fit operation here to find out if it negatively impacts performance in the common case (by requiring a reallocation + memcpy).